pkg/controller/operandrequest/reconcile_operand.go

```diff
@@ -134,21 +134,32 @@ func (r *ReconcileOperandRequest) reconcileOperand(requestKey types.NamespacedNa
 }
 
 // getCSV retrieves the ClusterServiceVersion
-func (r *ReconcileOperandRequest) getClusterServiceVersion(subName, subNamespace string) (*olmv1alpha1.ClusterServiceVersion, error) {
+func (r *ReconcileOperandRequest) getClusterServiceVersion(subName, packageName, subNamespace string) (*olmv1alpha1.ClusterServiceVersion, error) {
 	klog.V(3).Infof("Looking for the ClusterServiceVersion for Subscription %s in the namespace %s", subName, subNamespace)
 	sub, err := r.olmClient.OperatorsV1alpha1().Subscriptions(subNamespace).Get(subName, metav1.GetOptions{})
+
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Error("Failed to list Subscriptions: ", err)
+		return nil, err
+	}
+
 	if errors.IsNotFound(err) {
 		klog.V(3).Infof("There is no Subscription %s in the namespace %s", subName, subNamespace)
-		return nil, nil
+		sub, err = r.olmClient.OperatorsV1alpha1().Subscriptions(subNamespace).Get(packageName, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			klog.Error("Failed to list Subscriptions: ", err)
+			return nil, err
+		}
 	}
-	if err != nil {
-		klog.Error("Failed to list Subscriptions: ", err)
-		return nil, err
+
+	if errors.IsNotFound(err) {
```

Comment:
Seems this `errors.IsNotfound` duplicate with above

Reply:
The first `errors.IsNotfound` is for checking the subscription with `operator.name`. The second one is for checking the subscription with the operator package name.

Reply:
Looks a bit confused, can you create a method to get the sub with `operator.name` and package name?

Reply:
Yes, sure